## Dans ces bout de code, nous allons déterminer les huits types de service qui représente au total 99% des appels.

In [1]:
#1) Importation des packages
import pandas as pd
import os
from pathlib import Path

## Les 12 fichiers réprésentants les call pours les 12 mois pour l'année 2014, sont mis dans le dossier dataset. Ces fichiers sont d'extension.csv. Nous allons parcourir le dossier avec le package os de Python. Nous concatenons tout les content des 12 fichiers en un seul dataframe.


In [2]:
pathlist = Path("./dataset/").glob('*.csv')
df_all_month=pd.DataFrame()
total_line = 0 
for path in pathlist:
    df_call_month = pd.read_csv(path)
    shape = df_call_month.shape
    total_line =shape[0] + total_line
    df_all_month = pd.concat([df_all_month, df_call_month], axis=0)

# L'objectif de total_line est de le comparer au nombre de ligne que contiendra df_all_month
print("Nombre d'appel reçu par le centre d'appel au cours des 12 mois de l'année 2014 est : ", total_line)

Nombre d'appel reçu par le centre d'appel au cours des 12 mois de l'année 2014 est :  1543164


## Exploration de df_all_month

In [3]:
count_line =  (df_all_month.shape)[0]
if count_line==total_line:
    print("Le nombre de ligne de df_all_month est : ", count_line, ". Valeur qui est conforme à total_line. La concaténation s'est passée donc avec succès")
else :
    print("Le nombre de ligne de df_all_month est : ", count_line, ". Valeur qui n'est pas conforme à total_line : " , total_line ," . La concaténation ne s'est pas bien passée.")
    

Le nombre de ligne de df_all_month est :  1543164 . Valeur qui est conforme à total_line. La concaténation s'est passée donc avec succès


In [4]:
df_all_month.head()

date_received  queue_name  agent_number             answered consult  \
0  2014-01-02 08:03:21       30175        6935.0  2014-01-02 08:03:22     NaN   
1  2014-01-02 08:04:37       30560        6940.0  2014-01-02 08:04:38     NaN   
2  2014-01-02 08:06:26       30172        6968.0  2014-01-02 08:06:27     NaN   
3  2014-01-02 08:08:07       30175        9424.0  2014-01-02 08:08:08     NaN   
4  2014-01-02 08:08:26       30172        8076.0  2014-01-02 08:08:27     NaN   

  transfer               hangup  
0      NaN  2014-01-02 08:05:54  
1      NaN  2014-01-02 08:09:49  
2      NaN  2014-01-02 08:15:17  
3      NaN  2014-01-02 08:10:37  
4      NaN  2014-01-02 08:14:59

## Détermination du nombre d'appel reçu par service

In [5]:
# Nous faisons un group by, puis un count sur "Date_received".
# A cette sortie nous appliquons un trie par ordre décroissant sur date_received.
calls_by_service =  pd.DataFrame( df_all_month.groupby("queue_name")["date_received"].count()).sort_values(by = 'date_received', ascending=False)
# rénitialisation d'indice pour avoir transformer l'index réprésenter par queue_name en une colonne
calls_by_service = calls_by_service.reset_index()
# Nous renommons les colonnes  (optionnel)
calls_by_service.columns=['queue_name','call_received']
# Affichage 
calls_by_service 

queue_name  call_received
0        30175         603337
1        30560         329299
2        30172         285088
3        30181         157585
4        30179          68905
5        30066          38692
6        30518          31186
7        30241          10639
8        30511           6000
9        30584           2801
10       30519           2695
11       30729           1615
12       30764           1604
13       30747           1294
14       30363           1147
15       30176            701
16       30180            210
17       30177            121
18       30173             76
19       30325             47
20       30694             30
21       30174             26
22       30236             24
23       30598             21
24       30334             14
25       30178              6
26       30170              1

## Au terme de ce notebook, nous remarquons effectivement que nous avons 27 types de service. Les 99% d'appel représente les appels reçu par les top 8 des services les plus appelés. 
## Voici les Top 8 des services les plus appelés : 
## <p style = "color:red"> 30175, 30560, 30172, 30181, 30179, 30066, 30518, 30241 </p>

### <i> Amusement</i>

In [6]:
# Vérifions l'hypothèse : 99% représente les Top 8
all_call_received = (df_all_month.shape)[0]

call_Top_8=0
for i in range (8): 
    call_Top_8 = calls_by_service.loc[i, "call_received"] + call_Top_8

percent_call_top_8 = (call_Top_8/all_call_received)*100
print ("Le nombre total de call reçu par les Top 8 des services les plus utilisés est : ",call_Top_8,". Soit", percent_call_top_8, "%.")
print("Il repreésente à l'arrondi : ", round(percent_call_top_8) , "% du total des call reçu par le centre d'appel.")

print("\n \nYouuuuuuppiiiiiiiiiiiiii l'hypothèse est vérifiée.")

Le nombre total de call reçu par les Top 8 des services les plus utilisés est :  1524731 . Soit 98.80550609008505 %.
Il repreésente à l'arrondi :  99 % du total des call reçu par le centre d'appel.

 
Youuuuuuppiiiiiiiiiiiiii l'hypothèse est vérifiée.
